apps/prairielearn/src/lib/db-types.test.ts

```diff
@@ -0,0 +1,267 @@
+import _ from 'lodash';
+import { afterAll, beforeAll, describe, it } from 'vitest';
+import { type z } from 'zod';
+
+import { describeDatabase } from '@prairielearn/postgres-tools';
+
+import * as helperDb from '../tests/helperDb.js';
+
+import {
+  AccessLogSchema,
+  AccessTokenSchema,
+  AdministratorSchema,
+  AiGradingJobSchema,
+  AiQuestionGenerationPromptSchema,
+  AlternativeGroupSchema,
+  AssessmentAccessRuleSchema,
+  AssessmentInstanceSchema,
+  AssessmentModuleSchema,
+  AssessmentQuestionRolePermissionsSchema,
+  AssessmentQuestionSchema,
+  AssessmentSchema,
+  AssessmentScoreLogSchema,
+  AssessmentSetSchema,
+  AssessmentStateLogSchema,
+  AuditLogSchema,
+  AuthnProviderSchema,
+  BatchedMigrationJobSchema,
+  BatchedMigrationSchema,
+  ChunkSchema,
+  ClientFingerprintSchema,
+  CourseInstanceAccessRuleSchema,
+  CourseInstancePermissionSchema,
+  CourseInstanceRequiredPlanSchema,
+  CourseInstanceSchema,
+  CourseInstanceUsageSchema,
+  CoursePermissionSchema,
+  CourseRequestSchema,
+  CourseSchema,
+  CronJobSchema,
+  CurrentPageSchema,
+  DraftQuestionMetadataSchema,
+  EnrollmentSchema,
+  ExamModeNetworkSchema,
+  ExamSchema,
+  FeatureGrantSchema,
+  FileEditSchema,
+  FileSchema,
+  FileTransferSchema,
+  GlueCourseSchema,
+  GraderLoadSchema,
+  GradingJobSchema,
+  GroupConfigSchema,
+  GroupLogSchema,
+  GroupRoleSchema,
+  GroupSchema,
+  GroupUserRoleSchema,
+  GroupUserSchema,
+  InstanceQuestionSchema,
+  InstitutionAdministratorSchema,
+  InstitutionAuthnProviderSchema,
+  InstitutionSchema,
+  IssueSchema,
+  JobSchema,
+  JobSequenceSchema,
+  LastAccessSchema,
+  Lti13AssessmentsSchema,
+  Lti13CourseInstanceSchema,
+  Lti13InstanceSchema,
+  Lti13UserSchema,
+  LtiCredentialsSchema,
+  LtiLinkSchema,
+  LtiOutcomeSchema,
+  MigrationSchema,
+  NamedLockSchema,
+  NewsItemNotificationSchema,
+  NewsItemSchema,
+  PageViewLogSchema,
+  PlanGrantSchema,
+  QueryRunSchema,
+  QuestionGenerationContextEmbeddingSchema,
+  QuestionSchema,
+  QuestionScoreLogSchema,
+  QuestionTagSchema,
+  ReservationSchema,
+  RubricGradingItemSchema,
+  RubricGradingSchema,
+  RubricItemSchema,
+  RubricSchema,
+  SamlProviderSchema,
+  ServerLoadSchema,
+  SharingSetCourseSchema,
+  SharingSetQuestionSchema,
+  SharingSetSchema,
+  StripeCheckoutSessionSchema,
+  SubmissionGradingContextEmbeddingSchema,
+  SubmissionSchema,
+  TagSchema,
+  TimeSeriesSchema,
+  TopicSchema,
+  UserSchema,
+  UserSessionSchema,
+  VariantSchema,
+  VariantViewLogSchema,
+  WorkspaceHostLogSchema,
+  WorkspaceHostSchema,
+  WorkspaceLogSchema,
+  WorkspaceSchema,
+  ZoneSchema,
+} from './db-types.js';
+
+// Mapping from database table names to their corresponding Zod schemas
+const TABLE_SCHEMA_MAP: Record<string, z.ZodObject<any>> = {
```

Comment:
Thoughts on automating this mapping based on the name, instead of using an object? Mainly:
* If table ends with `s`, remove the s;
* Convert snake_case to PascalCase (capitalize first character plus any character that follows an underline);
* Attach `Schema` to end;
* Deal with individual exceptions (courses, pl_courses);
* Retrieve the schema from a namespace object instead of named imports.

Reply:
I agree, and made this change in 3d3e3d765d6f7b23f8c517710b90d776bb0c4807.